lib/shopify_app/login_protection.rb

```diff
@@ -54,7 +54,7 @@ def close_session
     def main_or_engine_login_url(params = {})
       main_app.login_url(params)
     rescue NoMethodError
-      shopify_app.login_url(params)
+      shopify_app_engine.login_url(host: "https://#{request.host}", params: params)
```

Comment:
Does this work in local environments without a port? For example, if my dev server is listening on port 3000 `myapp.com:3000` this would redirect to `myapp.com/app-name/login